Branch-SDK/src/io/branch/referral/network/BranchRemoteInterface.java

```diff
@@ -0,0 +1,276 @@
+package io.branch.referral.network;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import io.branch.referral.Branch;
+import io.branch.referral.BranchError;
+import io.branch.referral.BuildConfig;
+import io.branch.referral.Defines;
+import io.branch.referral.PrefHelper;
+import io.branch.referral.ServerResponse;
+
+/**
+ * <p/>
+ * Abstract class for Branch remote interface. This class provides the abstraction layer for network operations between
+ * Branch SDK and remote Branch servers. Extend this class for creating custom network interface.
+ * Class provide abstract method for implementing RESTful requests to Branch server
+ *
+ * @see {@link #doRestfulGet(String)}
+ * {@link #doRestfulPost(String, JSONObject)}
+ * {@link io.branch.referral.network.BranchRemoteInterface.BranchResponse}
+ * {@link BranchRemoteException}
+ * </p>
+ */
+public abstract class BranchRemoteInterface {
+    private static final String BRANCH_KEY = "branch_key";
+    public static final String RETRY_NUMBER = "retryNumber";
```

Comment:
It looks like only retry number is used in our implementation. Can you add a little documentation on the requirements of when to add it in?